target/linux/mvebu/files/arch/arm/boot/dts/armada-380-sa-w2.dts

```diff
@@ -0,0 +1,418 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
```

Comment:
Shouldn't be this upstreamed?

Reply:
I don't have any plans to send this for now because I don't have time to maintain dts in upstream...

Reply:
So, we should have time here to maintain it for you? :shrug:

Reply:
> I don't have any plans to send this for now because I don't have time to maintain dts in upstream...

After dts is accepted there is no need to maintain it upstream, it will be updated by target maintainer or if bindings to driver change, by the sole author of those changes. Unless You forgot something, then after sending changes it's usually accepted without questions.

> In fact,
> 
>     * there are already some device dts files in OpenWrt, I decided sending to upstream is not required

Yes there are and sending upstream is not required, I think.

>     * this dts has some OpenWrt-specific properties/nodes (`led-*` aliases, `firmware` partition, etc...), so I felt that the advantage of sending to upstream at this point is not very big

Yes these are OpenWrt specific, we do patch those in upstream dts and this is fine. Let me give You an example of possible breakage when dts is "out-of-date". At some point, can't remember where was the conversation, kernel bindings of a NAND controller changed and Linksys WRT 32x user reported no rootfs. Guess why only him and not the rest of Linksys WRT series? Because WRT 32x is the only one not upstreamed, the rest of WRT series was fine because author responsible for the changes also adjusted dts. Leaving aside the maintenance burden of changing downstream dts, imagine situation when someone only uses stable releases and that issue slipping through, because no one tested the snapshot. Also imagine possible hurdles recovering from that for casual user. Fortunately that WRT 32x issue was noticed before PR was merged. So please consider upstreaming.

Reply:
Thank you for the great explanation, I was convinced.
OK, I'll try to send dts to upstream...